rfd/0169-auto-updates-linux-agents.md

```diff
@@ -0,0 +1,926 @@
+  "agent_update_jitter_seconds": 10
+}
+```
+Notes:
+- Agents will only update if `agent_autoupdate` is `true`, but new installations will use `agent_version` regardless of the value in `agent_autoupdate`.
+- The edition served is the cluster edition (enterprise, enterprise-fips, or oss), and cannot be configured.
+- The host UUID is read from `/var/lib/teleport/host_uuid` by the updater.
+- The group name is read from `/var/lib/teleport/versions/updates.yaml` by the updater.
+
+### Teleport Resources
+
+#### Scheduling
+
+```yaml
+kind: autoupdate_config
+spec:
+  # agent_autoupdate allows turning agent updates on or off at the
+  # cluster level. Only turn agent automatic updates off if self-managed
+  # agent updates are in place.
+  agent_autoupdate: true|false
+
+  # agent_schedules contains both "regular" and "critical" schedules.
+  # The schedule used is determined by the agent_version_schedule associated
+  # with the version in autoupdate_version.
+  # Groups are not configurable with the "immediate" schedule.
+  agent_schedules:
+    # schedule is "regular" or "critical"
+    regular:
+      # name of the group
+    - name: staging-group
+      # days specifies the days of the week when the group may be updated.
+      #  default: ["*"] (all days)
+      days: [“Sun”, “Mon”, ... | "*"]
+      # start_hour specifies the hour when the group may start upgrading.
+      #  default: 0
+      start_hour: 0-23
+      # jitter_seconds specifies a maximum jitter duration after the start hour.
+      # The agent updater client will pick a random time within this duration to wait to update.
+      #  default: 0
+      jitter_seconds: 0-60
+      # timeout_seconds specifies the amount of time, after the specified jitter, after which
+      # an agent update will be considered timed out if the version does not change.
+      #  default: 60
+      timeout_seconds: 30-900
+      # failure_seconds specifies the amount of time after which an agent update will be considered
+      # failed if the agent heartbeat stops before the update is complete.
+      #  default: 0
+      failure_seconds: 0-900
+      # max_in_flight specifies the maximum number of agents that may be updated at the same time.
+      #  default: 100%
+      max_in_flight: 0-100%
+      # max_timeout_before_halt specifies the percentage of clients that may time out before this group
+      # and all dependent groups are halted.
+      #  default: 10%
+      max_timeout_before_halt: 0-100%
+      # max_failed_before_halt specifies the percentage of clients that may fail before this group
+      # and all dependent groups are halted.
+      #  default: 0
+      max_failed_before_halt: 0-100%
+      # requires specifies groups that must pass with the current version before this group is allowed
+      # to run using that version.
+      requires: ["test-group"]
+  # ...
+```
+
+Dependency cycles are rejected.
+Dependency chains longer than a week will be rejected.
```

Comment:
This restriction is to prevent configurations that have confusing behavior with a week-based schedule, not to enforce faster updates. E.g., monday-group depends on tuesday-group depends on wednesday-group, etc.

Restricting to less than a week wouldn't necessarily ensure critical updates happen faster, since a critical update could be released a week before the first scheduled time. Maybe our documentation and defaults for schedules could always use daily schedules for critical updates, to encourage faster rollouts?